app/Http/Middleware/CustomApiTokenAuth.php

```diff
@@ -0,0 +1,178 @@
+                return $next($request);
+            }
+        }
+        
+        // STEP 3: Try to authenticate using various methods
+        $token = $this->extractToken($request);
+        
+        if ($token) {
+            $user = User::where('api_token', $token)->first();
+            
+            if ($user) {
+                // Authenticate the user
+                Auth::login($user);
+                
+                // Set Authorization header for downstream middleware
+                $request->headers->set('Authorization', 'Bearer ' . $token);
+                
+                Log::debug("API Authentication successful", [
+                    'user_id' => $user->id,
+                    'token_source' => $this->tokenSource
+                ]);
+                
+                return $next($request);
+            } else {
+                Log::warning("Invalid API token", [
+                    'token_length' => strlen($token),
+                    'token_source' => $this->tokenSource
+                ]);
+            }
+        }
+        
+        // STEP 4: If we've reached this point, authentication failed
+        Log::warning("API Authentication failed", [
+            'path' => $request->path(),
+            'attempted_methods' => [
+                'session' => Auth::check(),
+                'cookie' => $request->hasCookie('restarters_apitoken'),
+                'header' => $request->hasHeader('Authorization'),
+                'query' => $request->has('api_token'),
+            ]
+        ]);
+        
+        // For AJAX or API requests, return a JSON response
+        if ($request->expectsJson() || $request->is('api/*')) {
+            return response()->json([
+                'message' => 'Unauthenticated.',
+                'debug' => [
+                    'token_source_tried' => $this->tokenSource,
+                    'session_auth' => Auth::check(),
+                    'cookie_auth_attempted' => $request->hasCookie('restarters_apitoken'),
+                    'header_auth_attempted' => $request->hasHeader('Authorization'),
+                    'query_auth_attempted' => $request->has('api_token'),
+                ]
+            ], 401);
+        }
+        
+        // For web requests, redirect to login page
+        throw new AuthenticationException(
+            'Unauthenticated.',
+            ['api'],
+            $request->expectsJson() ? null : route('login')
+        );
+    }
+    
+    /**
+     * The source of the token that was used
+     */
+    private $tokenSource = 'none';
+    
+    /**
+     * Try all possible methods to extract an API token from the request
+     */
+    private function extractToken(Request $request)
+    {
+        // Method 1: Check for token in query string
+        $token = $request->query('api_token');
+        if (!empty($token)) {
+            $this->tokenSource = 'query';
+            Log::debug("Found token in query string", ['length' => strlen($token)]);
```

Comment:
Do we want to keep all of these debug statements?

Reply:
I was planning on removing it but I am going to come back to this file in the future to properly address the authentication problem and see about moving to Laravel Sanctum or Laravel Fortify. I could just re-add debug statements at that point in time but I might forget what I was trying to do originally.